crypto/fipsmodule/rsa/rsa.c

```diff
@@ -903,6 +903,42 @@ DEFINE_LOCAL_DATA(BIGNUM, g_small_factors) {
   out->flags = BN_FLG_STATIC_DATA;
 }
 
+static int EVP_RSA_KEY_check_fips(RSA *key) {
+  uint8_t msg[1] = {0};
+  size_t msg_len = 1;
+  int ret = 0;
+  uint8_t* sig_der = NULL;
+  EVP_PKEY *evp_pkey = EVP_PKEY_new();
+  EVP_MD_CTX *ctx = EVP_MD_CTX_new();
```

Comment:
We could skip this allocation if you create a EVP_MD_CTX on the stack then call EVP_MD_CTX_init on it.

Reply:
Added and applied the same change to the EC PWCT

Reply:
Does the key get copied to the context? It may be desirable to zeroise it if yes.

Reply:
It's an MD context. I guess it would have nothing to do with the key (nor the scalar k in ECDSA).

Reply:
Just added `EVP_MD_CTX_cleanse` to zeroize it anyways, just to be safe :)